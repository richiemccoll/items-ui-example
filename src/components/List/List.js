import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from 'styled-components';
import Card from "../Card/Card";

const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;

    @media (min-width: 700px) {
        flex-direction: row;
        flex-flow: row wrap;
    }
`;

class List extends Component {

  render() {
    const { data, toggleForm = null } = this.props;

    return (
      <Wrapper>
        {data.map((cake, index) => {
          return (
            <Card
              key={Math.random().toString(36).slice(2)}
              id={cake.title + index}
              title={cake.title}
              image={cake.image}
              description={cake.desc}
              toggleForm={toggleForm}
            />
          );
        })}
      </Wrapper>
    );
  }
}

export default List;

List.PropTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  toggleForm: PropTypes.func
};
