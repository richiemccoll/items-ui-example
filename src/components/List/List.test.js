import React from "react";
import List from "./List";
import renderer from "react-test-renderer";

describe("List Component", () => {
  const listProps = {
    data: [
      {
        title: "test",
        description: "test",
        image: ""
      },
      {
        title: "test2",
        description: "test2",
        image: ""
      }
    ],
    toggleForm() {
      return;
    }
  };

  it("should render correctly based on props", () => {
    const component = renderer.create(
      <List data={listProps.data} toggleForm={listProps.toggleForm} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
