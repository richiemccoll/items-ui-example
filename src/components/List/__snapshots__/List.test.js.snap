// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component should render correctly based on props 1`] = `
<div
  className="sc-ifAKCX egvpF"
>
  <div
    className="sc-bdVaJa iyRwOl"
  >
    <div
      className="sc-bwzfXH gOHCgz"
    >
      <img
        alt="test"
        className="sc-htpNat emEaka"
        src=""
      />
    </div>
    <div
      className="sc-bxivhb gutIee"
    >
      <h2
        className="ui header"
      >
        test
      </h2>
      <h3
        className="ui header"
      />
      <button
        className="ui green basic button"
        onClick={[Function]}
        tabIndex={undefined}
      >
        Edit
      </button>
    </div>
  </div>
  <div
    className="sc-bdVaJa iyRwOl"
  >
    <div
      className="sc-bwzfXH gOHCgz"
    >
      <img
        alt="test2"
        className="sc-htpNat emEaka"
        src=""
      />
    </div>
    <div
      className="sc-bxivhb gutIee"
    >
      <h2
        className="ui header"
      >
        test2
      </h2>
      <h3
        className="ui header"
      />
      <button
        className="ui green basic button"
        onClick={[Function]}
        tabIndex={undefined}
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;
